src/hotspot/cpu/aarch64/aarch64_sve.ad

```diff
@@ -3305,3 +3305,67 @@ instruct insertL(vReg dst, vReg src, iRegL val, immI idx, vReg tmp, vReg tmp2, p
   %}
   ins_pipe(pipe_slow);
 %}
+
+instruct insertF(vReg dst, vReg src, vRegF val, immI idx, vReg tmp, vReg tmp2, pRegGov pTmp, rFlagsReg cr)
+%{
+  predicate(UseSVE > 0 &&
+            n->bottom_type()->is_vect()->element_basic_type() == T_FLOAT);
+  match(Set dst (VectorInsert (Binary src val) idx));
+  effect(TEMP tmp, TEMP tmp2, TEMP pTmp, KILL cr);
+  ins_cost(5 * SVE_COST);
+  format %{ "sve_index $tmp, S, 0, 1\n\t"
+            "sve_dup $tmp2, S, $idx\n\t"
+            "sve_cmpeq $pTmp, $tmp, $tmp2\n\t"
+            "sve_orr $dst, $src, $src\n\t"
+            "sve_cpy $dst, $pTmp, $val\n\t# insert into vector (F)" %}
+  ins_encode %{
+    Assembler::SIMD_RegVariant size =
+              elemType_to_regVariant(vector_element_basic_type(this));
+    __ sve_index(as_FloatRegister($tmp$$reg), __ S, 0, 1);
+    __ sve_dup(as_FloatRegister($tmp2$$reg), __ S, (int)($idx$$constant));
+    __ sve_cmpeq(as_PRegister($pTmp$$reg), size, ptrue,
```

Comment:
SVE support immedicate CMP: `CMPEQ <Pd>.<T>, <Pg>/Z, <Zn>.<T>, #<imm>`. Could you please use the imm version here? We can save one `dup` instruction here.

Reply:
Sure. Thank you for your wonderful advice.

Reply:
> SVE support immedicate CMP: `CMPEQ <Pd>.<T>, <Pg>/Z, <Zn>.<T>, #<imm>`. Could you please use the imm version here? We can save one `dup` instruction here.

I have remembered why I choose this `cmpeq`. If you use `immedicate CMP`, the `# <imm>` is in range `[-16,15]`. I think we should use `cmpeq` in my codes.

Reply:
OK, so a vector CMP is needed here. Thanks for the explanation!

Reply:
Maybe you can check whether idx is in CMPEQ imm range to generate different code?

Reply:
> Maybe you can check whether idx is in CMPEQ imm range to generate different code?

I add a fast path by using `CMPEQ imm`. 

I shift the range of idx from [0,31] to [-16,15] to suit the range limit of `imm5`.